torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1103,11 +1103,8 @@ void ProcessGroupNCCL::abortCommsFromMap(
   for (auto& it : ncclCommsMap) {
     auto& devName = it.first;
     auto& ncclComm = it.second;
-    at::cuda::OptionalCUDAGuard gpuGuard;
     at::DeviceIndex deviceIndex = getIndexFromDeviceKey(devName);
```

Comment:
For P2P comms, the deviceIndex could be -1 (invalid), as the keys in the map could be non deviceIndex, but rank to rank numbers.

Reply:
Thanks for the information. Let me revert this change here and add the above comments into the code.